src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DeleteOperation.java

```diff
@@ -12,29 +12,68 @@
 import io.stargate.sgv2.jsonapi.service.operation.model.ReadOperation;
 import io.stargate.sgv2.jsonapi.service.shredding.model.DocumentId;
 import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Supplier;
 
 /**
  * Executes readOperation to get the documents ids based on filter condition. All the ids are
  * deleted as LWT based on the id and tx_id.
  */
-public record DeleteOperation(CommandContext commandContext, ReadOperation readOperation)
+public record DeleteOperation(
+    CommandContext commandContext,
+    ReadOperation readOperation,
+    /**
+     * Added parameter to pass number of document to be deleted, this is needed because read
+     * documents limit changed to deleteLimit + 1
+     */
+    int deleteLimit)
     implements ModifyOperation {
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
-    Uni<ReadOperation.FindResponse> docsToDelete = readOperation().getDocuments(queryExecutor);
+    final AtomicBoolean boolenFlag = new AtomicBoolean(false);
     final QueryOuterClass.Query delete = buildDeleteQuery();
+    final Multi<ReadOperation.FindResponse> findResponses =
+        Multi.createBy()
+            .repeating()
+            .uni(
+                () -> new AtomicReference<String>(null),
+                stateRef -> {
+                  Uni<ReadOperation.FindResponse> docsToDelete =
+                      readOperation().getDocuments(queryExecutor, stateRef.get());
+                  return docsToDelete
+                      .onItem()
+                      .invoke(findResponse -> stateRef.set(findResponse.pagingState()));
+                })
+            .whilst(findResponse -> findResponse.pagingState() != null);
```

Comment:
:warning: 

This is not good, this will fetch until all documents are collected.. This handling below is questionable, as it works on incoming items.. Thus, you will effectively fetch all the documents and in this transformation you will limit them.. Instead you should do:

* fetch as now in the loop
* expand documents
* and then use `.select().first(deleteLimit)`

This will ensure that the fetching stops as soon as you have needed amount of elements

Reply:
There should be a test ensuring this, I see no such test.

Reply:
> * and then use `.select().first(deleteLimit)`

@ivansenic ReadOperation used is set with a read limit. Maximum records we read is (deleteLimit +1).  Additional 1 record is read to correctly set the moreData flag. Handling on count in DeleteOperation class is to make sure the additional record is not deleted. Since collect() is done only at the end to get deleted Ids. 
Delete limit is passed from DeleteManyCommandResolver.getFilteringOption().
I added this as comment to deleteLimit variable in DeleteOperation class.

> There should be a test ensuring this, I see no such test.

I have created a test here which will make sure we delete only required count of documents. DeleteManyIntegrationTest.DeleteMany.deleteManyNoFilterMoreDataFlag(). FindOperationTest unit methods should confirm that limit provided to read operation is applied to Select query.

Reply:
It's not about how many documents is deleted, it's how many is fetched. The statement:

```
.whilst(findResponse -> findResponse.pagingState() != null);
```
Will go and re-execute the read command, you will keep fetching the documents, but not passing after that bottom observer..

In addition, this whole logic is way to complex and could be simplified by expanding docs and taking first `deleteLimit +1` elements.. There you know there is at least one extra document and can delete the first 20.. If you want I can design a test to prove you are making unnecessary queries here to the bridge?

Reply:
The read operation is created with limit deleteLimit +1. Let's say required deleteLimit is 20, read operation is set with limit of 21. This will translate to CQL as `select key, tx_id from collection where <condition> LIMIT 21`
The data fetch by the read operation will stop once 21 document ids are read from the table. 

> In addition, this whole logic is way to complex and could be simplified by expanding docs and taking first `deleteLimit +1` elements..

What is suggested is what been done. The whilst conditions is there to satisfy when deleteLimit is greater than read pageSize.

If read pageSize is 20 and deleteLimit is 100. The whilst condition will make it iterate to fetch 20 document ids at a time.